starter-code/src/viking.js

```diff
@@ -1,11 +1,111 @@
 // Soldier
-function Soldier() {}
+function Soldier(health, strength) {
+    this.health=health;
+    this.strength=strength;
+}
+
+Soldier.prototype.attack=function(){
+    return this.strength;
+}
+Soldier.prototype.receiveDamage=function(damage){
+    this.health-=damage;
+}
 
 // Viking
-function Viking() {}
+function Viking(name,health,strength) {
+    Soldier.call(this,health,strength);
+    this.name=name;
+    
+}
+
+Viking.prototype = Object.create(Soldier.prototype);
+Viking.prototype.constructor = Viking;
+Viking.prototype.battleCry=function(){
+    return "Odin Owns You All!";
+}
 
+Viking.prototype.receiveDamage=function(damage){
+    this.health-=damage;
+    if(this.health-damage>0){
+        return this.name+" has received "+damage +" points of damage";
+    }else {
+        return this.name+" has died in act of combat";
+    }
+}
 // Saxon
-function Saxon() {}
+function Saxon(health,strength) {
+    Soldier.call(this,health,strength);
+}
+
+Saxon.prototype= Object.create(Soldier.prototype);
+Saxon.prototype.receiveDamage=function(damage){
+    this.health-=damage;
+    if(this.health>0){
+        return "A Saxon has received "+damage +" points of damage";
+    }else {
+        return "A Saxon has died in combat";
+    }
+}
 
 // War
-function War() {}
+function War() {
+    this.vikingArmy=[];
+    this.saxonArmy=[];
+}
+function generateViking () {
+    var name     = "Harald";
+    var strength = 150;
+    var health   = 300;
+
+    return new Viking(name, health, strength);
+  }
+
+  function generateSaxon () {
+    var health   = 60;
+    var strength = 25;
+    return new Saxon(health, strength);
+  }
```

Comment:
from line 55 to line 67 are not needed, because you already have your variables before and you don't need to generate anything using manual entries. When you put manual entries on the property it will always return those values regardless of the arguments you give the function

Reply:
I agree. I put it because i was working in repl to test the values...Now I remove them